paasta_tools/contrib/kill_bad_containers.py

```diff
@@ -0,0 +1,78 @@
+#!/opt/venvs/paasta-tools/bin/python3
+import sys
+
+import iptc
+
+from paasta_tools import iptables
+from paasta_tools.utils import get_docker_client
+
+
+def get_container_from_dport(dport, docker_client):
+    for container in docker_client.containers():
+        try:
+            ports = container['Ports']
+            for port in ports:
+                if "PublicPort" in port:
+                    if port["PublicPort"] == int(dport):
+                        return container
+        except KeyError:
+            print(ports)
+            pass
+
+
+def target_rule_to_dport(rule):
+    try:
+        # (('tcp', (('dport', ('31493',)),)),)
```

Comment:
fun fact: this comment is actually almost valid python destructuring code:

    (_tcp,((_dport,(value,),),),), = rule.matches
    return int(value)

not that i suggest to use it

Reply:
Interesting, I hadn't thought about going in that direction (Because I copy/pasted it out of a repl).

I mean, why *not* do it like that? (still with a try/except though)

Reply:
i think it doesn't really make the structure of right-hand side obvious, which is what i would want from pattern-matching/destructuring assignments